tests/inc/clean-emails-test.php

```diff
@@ -68,6 +68,23 @@ public function data_comment_email_headers() {
 				'headers'  => '',
 				'expected' => "Content-Type: text/html; charset=\"UTF-8\"\n",
 			],
+			'header string with headers, but without content type header' => [
+				'headers'  => 'From: "Blogname" <blogname@blogdomain.com>
+Reply-To: "comment_author@theirdomain.com" <comment_author@theirdomain.com>
+',
+				'expected' => 'From: "Blogname" <blogname@blogdomain.com>
+Reply-To: "comment_author@theirdomain.com" <comment_author@theirdomain.com>
+Content-Type: text/html; charset="UTF-8"
+',
+			],
+			'header string with headers, but without content type header and missing new line at end of headers' => [
+				'headers'  => 'From: "Blogname" <blogname@blogdomain.com>
+Reply-To: "comment_author@theirdomain.com" <comment_author@theirdomain.com>',
+				'expected' => 'From: "Blogname" <blogname@blogdomain.com>
+Reply-To: "comment_author@theirdomain.com" <comment_author@theirdomain.com>
+Content-Type: text/html; charset="UTF-8"
+',
+			],
```

Comment:
@d-claassen Those actually also contain a new line, just in code, not in text, i.e. the tested "phrases" in those tests have a `"\n" included.

For a simple one-line test case, that's still pretty readable, but as soon as the test data becomes more extended, concatenating the test input and/or using single quote/double quote magic becomes less and less readable, which is why I choose to use multi-line text format with the explicit new lines for the longer test cases.

Reply:
Hang on... I see what you mean. Yes, there are two where I didn't add the `"\n"`. Those are not testing for the new line, they are testing the call to `str_replace()` which doesn't care about the new line.

Generally speaking, those new lines are only "our" concern when we _add_ a header, not when we replace something _in_ an existing header (then those should have been added by the code which added the original header).

Reply:
Now this makes me wonder if we should even add the additional newline **after** the Content-Type header, since we're already ensuring it's on a newline by putting a newline **before** the Content-Type header.

Reply:
Yes, we should. The headers sting should always end on a new line, so if we add a line, we also need to add a newline after it.

Reply:
Ah, now I get it. That's actually the behavior the other test is asserting: an incoming header that's ending in a newline, should keep ending with a newline. Thanks for explaining 🙂